src/components/string_input.tsx

```diff
@@ -63,15 +55,12 @@ const StringInput: FC<StringInputProps> = ({
         <>
             <FloatingLabel className="mb-3" controlId={id} label={label}>
                 <Form.Control
-                    ref={ref}
-                    onChange={event => {
-                        setCursor(event.target.selectionStart) // Set the cursor position as the selectionStart
-                        handleChange(event.target.value)
-                    }}
+                    onChange={event => handleChange(event.target.value)}
                     type="text"
                     value={value || ''}
                     isInvalid={Boolean(error)}
                 />
+                <Form.Text>{hint}</Form.Text>
```

Comment:
Please check whether this still takes up vertical space when hint is empty. You may need to wrap this with conditional code:  `hint ? <Form.Text>{hint}</Form.Text> : null`

Reply:
Yes, it takes vertical space. I have added a null check as suggested. It works fine now.